src/box/sql/show.c

```diff
@@ -0,0 +1,425 @@
+	sql_desc_append(desc, " CONSTRAINT %s CHECK(%s)", constraint_name,
+			func->def->body);
+	sql_xfree(constraint_name);
+}
+
+/** Add a tuple check constraint to the statement description. */
+static void
+sql_describe_tuple_check(struct sql_desc *desc,
+			 const struct tuple_constraint_def *cdef)
+{
+	assert(cdef->type == CONSTR_FUNC);
+	const struct func *func = func_by_id(cdef->func.id);
+	if (func->def->language != FUNC_LANGUAGE_SQL_EXPR) {
+		sql_err_append(desc, "Check constraint", cdef->name);
+		return;
+	}
+	char *constraint_name = sql_escaped_name_new(cdef->name);
+	sql_desc_append(desc, ",\nCONSTRAINT %s CHECK(%s)", constraint_name,
+			func->def->body);
+	sql_xfree(constraint_name);
+}
+
+/** Add a field to the statement description. */
+static void
+sql_describe_field(struct sql_desc *desc, const struct field_def *field)
+{
+	char type_str[TYPE_STR_SIZE];
+	const char *type_str_original = field_type_strs[field->type];
+	size_t type_str_len = strlen(type_str_original);
+	assert(type_str_len < TYPE_STR_SIZE);
+	for (size_t i = 0; i < type_str_len; ++i)
+		type_str[i] = toupper(type_str_original[i]);
+	type_str[type_str_len] = '\0';
+	char *field_name = sql_escaped_name_new(field->name);
+	sql_desc_append(desc, "%s %s", field_name, type_str);
+	sql_xfree(field_name);
+
+	if (field->coll_id != 0) {
+		struct coll_id *coll_id = coll_by_id(field->coll_id);
+		if (coll_id == NULL) {
+			const char *id_str = tt_sprintf("%d", field->coll_id);
+			sql_err_append(desc, "Collation", id_str);
+		} else {
+			char *coll_name = sql_escaped_name_new(coll_id->name);
+			sql_desc_append(desc, " COLLATE %s", coll_name);
+			sql_xfree(coll_name);
+		}
+	}
+	if (!field->is_nullable)
+		sql_desc_append(desc, " NOT NULL");
+	if (field->default_value != NULL)
+		sql_desc_append(desc, " DEFAULT(%s)", field->default_value);
+	for (uint32_t i = 0; i < field->constraint_count; ++i) {
+		struct tuple_constraint_def *cdef = &field->constraint_def[i];
+		assert(cdef->type == CONSTR_FKEY || cdef->type == CONSTR_FUNC);
+		if (cdef->type == CONSTR_FKEY)
+			sql_describe_field_foreign_key(desc, cdef);
+		else
+			sql_describe_field_check(desc, field->name, cdef);
+	}
+}
+
+/** Add a primary key to the statement description. */
+static void
+sql_describe_primary_key(struct sql_desc *desc, const struct space *space)
+{
+	const struct index *pk = space->index[0];
+	assert(pk->def->opts.is_unique && pk->def->type == TREE);
+
+	bool has_sequence = false;
+	if (space->sequence != NULL) {
+		struct sequence_def *sdef = space->sequence->def;
+		if (sdef->step != 1 || sdef->min != 0 || sdef->start != 1 ||
+		    sdef->max != INT64_MAX || sdef->cache != 0 || sdef->cycle ||
+		    strcmp(sdef->name, space->def->name) != 0)
+			sql_err_append(desc, "Sequence", sdef->name);
+		else
+			has_sequence = true;
+	}
+
+	char *pk_name = sql_escaped_name_new(pk->def->name);
+	sql_desc_append(desc, ",\nCONSTRAINT %s PRIMARY KEY(", pk_name);
+	sql_xfree(pk_name);
```

Comment:
There are quite many of these triplets - name new + append + xfree. I would suggest to introduce `sql_desc_append_escape(desc, name)` which would the the same. It would force you to write sometimes not much less code, but you won't have to keep track of all those xfree:
```C
sql_desc_append(desc, ",\nCONSTRAINT ");
sql_desc_append_escape(desc, pk->def->name);
sql_desc_append(desc, "PRIMARY KEY(");
```
In some places it will turn into just 1 or 2 lines. Up to you.